otherlibs/stdune-unstable/temp.ml

```diff
@@ -58,22 +60,21 @@ let () =
 let temp_in_dir ?perms what ~dir ~prefix ~suffix =
   let path =
     let create = create ?perms what in
-    try_times 1000 ~f:(fun _ ->
-        let name = temp_file_name ~temp_dir:dir ~prefix ~suffix in
-        create name;
-        name)
-    |> Path.of_string
+    try_paths 1000 ~dir ~prefix ~suffix ~f:(fun path ->
+        create path;
+        path)
   in
   let set = set what in
   set := Path.Set.add !set path;
   path
 
 let create ?perms what ~prefix ~suffix =
-  let dir = Filename.get_temp_dir_name () in
+  let dir =
+    Filename.get_temp_dir_name () |> Path.of_filename_relative_to_initial_cwd
+  in
   temp_in_dir ?perms what ~dir ~prefix ~suffix
 
-let temp_in_dir ?perms what ~dir =
-  temp_in_dir ?perms what ~dir:(Path.to_absolute_filename dir)
+let temp_in_dir ?perms what ~dir = temp_in_dir ?perms what ~dir
```

Comment:
Do we still need this alias?

Reply:
Yes, we export this in `temp.mli` and still use in a couple of places.

Reply:
What's the point of the alias though? Previously we were doing a path conversion, now we're just doing `let x = x`.

Reply:
Btw, one of the two use sites, in `src/cache/local.ml`, will soon be gone. Another one is in `dune_engine/dtemp.mli` and I suspect we can get rid of it too.

Reply:
> What's the point of the alias though? Previously we were doing a path conversion, now we're just doing `let x = x`.

Oh, I got confused. Indeed, we no longer need this alias. Removing.

Reply:
Done!